src/main.ts

```diff
@@ -54,29 +56,42 @@ const runAction = async (config: Config): Promise<void> => {
     core.warning(`Failed to login to Pulumi service: ${result.stderr}`);
   }
 
-  const stackArgs: LocalProgramArgs = {
-    stackName: config.stackName,
-    workDir: workDir,
-  };
-
-  const stackOpts: LocalWorkspaceOptions = {};
+  const wsOpts: LocalWorkspaceOptions = {};
   if (config.secretsProvider != '') {
-    stackOpts.secretsProvider = config.secretsProvider;
+    wsOpts.secretsProvider = config.secretsProvider;
   }
 
-  const stack = await (config.upsert
-    ? LocalWorkspace.createOrSelectStack(stackArgs, stackOpts)
-    : LocalWorkspace.selectStack(stackArgs, stackOpts));
+  // Only initialize `stack` when the command is not `output`.
+  // When the command is `output` we want to avoid the underlying call to `pulumi stack select`,
+  // which requires a Pulumi.yaml file to be present.
```

Comment:
The only downside is that we aren't respecting `config.upsert` for the `output` command, so if the stack doesn't exist, it won't be created. I actually think this is ok and the behavior we want for `output`. You really only want to get stack outputs for a stack that exists.

If we felt it was important to keep `upsert` working for the `output` command, we could make a change below on line 131: when calling `outputs = await ws.stackOutputs(config.stackName);` we could wrap that in a try/catch and if it's a `StackNotFoundError` and `config.upsert` is true, we could create the stack and return empty outputs. My preference is to not do this unless we get feedback that it's needed for some scenario.